tensorflow/lite/experimental/shlo/ops/convolution_test.cc

```diff
@@ -0,0 +1,615 @@
+  Tensor rhs_dilation{.type = TensorType{.shape = shape_parametrs,
+                                         .element_type = DataType::kSI64},
+                      .data = rhs_dilation_values.data()};
+  Tensor window_reversal{.type = TensorType{.shape = shape_parametrs,
+                                            .element_type = DataType::kI1},
+                         .data = window_reversal_values.data()};
+  Tensor inputSpatialDimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = inputSpatialDimensions_values.data()};
+  Tensor kernel_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = kernel_spatial_dimensions_values.data()};
+  Tensor output_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = output_spatial_dimensions_values.data()};
+  Tensor output_tensor{
+      .type = QuantizedPerTensorTensorType{.shape = shape_result,
+                                           .element_type = tensor_type},
+      .data = output_data.data()};
+
+  absl::InlinedVector<PrecisionTypes, 2> precision_configs = {
+      PrecisionTypes::DEFAULT, PrecisionTypes::DEFAULT};
+  auto op = Create(ConvolutionOp::Attributes{
+      .window_strides = window_stride,
+      .padding = padding,
+      .lhs_dilation = lhs_dilation,
+      .rhs_dilation = rhs_dilation,
+      .window_reversal = window_reversal,
+      .input_batch_dimension = input_batch_dimension,
+      .input_feature_dimension = input_feature_dimension,
+      .input_spacial_dimensions = inputSpatialDimensions,
+      .kernel_input_feature_dimension = kernel_input_feature_dimension,
+      .kernel_output_feature_dimension = kernel_output_feature_dimension,
+      .kernel_spacial_dimensions = kernel_spatial_dimensions,
+      .output_batch_dimension = output_batch_dimension,
+      .output_feature_dimension = output_feature_dimension,
+      .output_spacial_dimensions = output_spatial_dimensions,
+      .feature_group_count = feature_group_count,
+      .batch_group_count = batch_group_count,
+      .precision_configs = precision_configs});
+  Vector<StorageT> expected_data =
+      Vector<StorageT>{1, 2, 3, 1, 2, 3, 1, 2, 3, 2};
+
+  Vector<StorageT> expected_data_quantized(shape_result.NumElements());
+  std::transform(expected_data.begin(), expected_data.end(),
+                 expected_data_quantized.begin(), [&](StorageT val) {
+                   return Quantize<TypeParam::kStorage, TypeParam::kExpressed>(
+                       static_cast<ExpressedT>(val), zero_point,
+                       static_cast<ExpressedT>(1.0) / scale);
+                 });
+
+  ASSERT_OK(Prepare(op, lhs, rhs, output_tensor));
+  ASSERT_OK(Evaluate(op, lhs, rhs, output_tensor));
+
+  constexpr double kEpsilon = 0.1;
+  EXPECT_THAT(output_data, Pointwise(FloatEq(), expected_data));
```

Comment:
Why is this using FloatEq for integer storage?

Reply:
changed to Eq() for integer storage(you can check it in PR#67245)